### PR TITLE
Improvement messaging

### PR DESCRIPTION
**Pull request by victor-paumier**
- Lastname mandatory
- Only let managers post in mentors / mentees channels
- Replace email by fullname
- Fix send message on thread with large number of participants
- Remove archived users from mentors/mentees channels
- Account archived on user dashboard
- Secure message route

### CODE REVIEW

File: assets/sass/pages/inbox.scss [hunk 1]
```diff
@@ -217,6 +217,12 @@ $border-color: #c4c4c4;
     height: 455px;
     overflow-y: auto;
     padding: 0 10px;
+
+    &--no-reply {
```
Comment (JeremieSamson) [inline assets/sass/pages/inbox.scss]: not sure but this syntax should be _ ?

File: src/Controller/MessageController.php [hunk 2]
```diff
@@ -27,8 +30,11 @@ public function inboxAction()
         if (\count($params['threads']) > 0) {
             $thread = $params['threads'][0];
             $this->get('fos_message.thread_reader')->markAsRead($thread);
-            $replyForm = $this->get('fos_message.reply_form.factory')->create($thread);
-            $params['form'] = $replyForm->createView();
+
+            if ($this->messageAuthorizer->canMessageThread($thread)) {
+                $replyForm = $this->get('fos_message.reply_form.factory')->create($thread);
```
Comment (JeremieSamson) [inline src/Controller/MessageController.php]: no injection with fos_message.reply_form.factory service ?

File: src/DataTable/DataTableModel.php [hunk 3]
```diff
@@ -54,7 +54,7 @@ public function setSearch(array $search = []): void
 
                 $searchValue = $column['search']['value'];
                 if ('' !== $searchValue) {
-                    $this->search[$columnName] = (false == strstr($searchValue, ','))
+                    $this->search[$columnName] = (false == strpos($searchValue, ','))
```
Comment (JeremieSamson) [inline src/DataTable/DataTableModel.php]:
```suggestion
                    $this->search[$columnName] = (false === strpos($searchValue, ','))
```

File: templates/Dashboard/index.html.twig [hunk 4]
```diff
@@ -43,6 +43,13 @@
                                 </p>
                             {% endif %}
 
+                            {% if user.isArchived %}
+                                <p class="account-status text-danger">
+                                    <img src="{{ asset('images/nok.svg') }}"
+                                         alt=""> {{ 'dashboard.profile.archived'|trans }}
```
Comment (JeremieSamson) [inline templates/Dashboard/index.html.twig]: is empty alt needed ?

File: assets/sass/pages/inbox.scss (hunk 1 above)
Comment (victor-paumier) [reply assets/sass/pages/inbox.scss]: _ is for element, -- is for modifier

**Pull request merged**